timemachine/cpp/src/kernels/k_nonbonded.cuh

```diff
@@ -306,6 +306,92 @@ double __device__ __forceinline__ fix_nvidia_fmad(double a, double b, double c,
 
 // } // 0 or 1, how much we offset from the plane by )
 
+
+// Compute the terms associated with electrostatics.
+// This is pulled out into a function to ensure that the same bit values
+// are computed to ensure that that the fixed point values are exactly the same regardless
+// of where the values are computed.
+template <
+    typename RealType,
+    bool COMPUTE_U
+>
+void __device__ __forceinline__ compute_electrostatics(
+    const RealType charge_scale,
+    const RealType qi,
+    const RealType qj,
+    const RealType d2ij,
+    const RealType beta,
+    RealType &dij,
+    RealType &inv_dij,
+    RealType &inv_d2ij,
+    RealType &ebd,
+    RealType &es_prefactor,
+    RealType &u
+) {
+    inv_dij = rsqrt(d2ij);
+
+    dij = d2ij*inv_dij;
+    inv_d2ij = inv_dij*inv_dij;
+
+    RealType qij = qi*qj;
+    es_prefactor = charge_scale*qij*inv_dij*real_es_factor(beta, dij, inv_d2ij, ebd);
+
+    if (COMPUTE_U) {
+        u = charge_scale*qij*inv_dij*ebd;
+    }
+}
+
+// Handles the computation related to the LJ terms.
+// This is pulled out into a function to ensure that the same bit values
+// are computed to ensure that that the fixed point values are exactly the same regardless
+// of where the values are computed.
+template <
+    typename RealType,
+    bool COMPUTE_U
+>
+void __device__ __forceinline__ compute_lj(
+    RealType lj_scale,
+    RealType eps_i,
+    RealType eps_j,
+    RealType sig_i,
+    RealType sig_j,
+    RealType inv_dij,
+    RealType inv_d2ij,
+    RealType &u,
+    RealType &delta_prefactor,
+    RealType &sig_grad,
+    RealType &eps_grad
+) {
+    RealType eps_ij = eps_i * eps_j;
+    RealType sig_ij = sig_i + sig_j;
+
+    RealType sig_inv_dij = sig_ij*inv_dij;
+    RealType sig2_inv_d2ij = sig_inv_dij*sig_inv_dij;
+    RealType sig4_inv_d4ij = sig2_inv_d2ij*sig2_inv_d2ij;
+    RealType sig6_inv_d6ij = sig4_inv_d4ij*sig2_inv_d2ij;
+    RealType sig6_inv_d8ij = sig6_inv_d6ij*inv_d2ij;
+    RealType sig5_inv_d6ij = sig_ij*sig4_inv_d4ij*inv_d2ij;
+
+    // optimize this a little more
```

Comment:
we can probably remove these comments now - I don't think they resulted in any speedsup